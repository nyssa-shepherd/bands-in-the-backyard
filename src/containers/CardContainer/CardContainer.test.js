import React from 'react';
import { shallow } from 'enzyme';
import { 
  mapStateToProps, 
  CardContainer } from './CardContainer.js';

describe('CardContainer', () => {
  let wrapper;
  let mockMatch;
  let mockArtists;

  beforeEach(() => {
    mockMatch = {path: '/artists'};
    mockArtists = [{
      name: 'name',
      city: 'city',
      state: 'state',
      date: 'date',
      venue: 'venue',
      ticketUrl: 'url'
    }];
    wrapper = shallow(<CardContainer artistInLocation={mockArtists} match={mockMatch} />);
  });

  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('map the store correctly', () => {
    const event1 = {
      image: 'url',
      name: 'name',
      city: 'city',
      state: 'state',
      date: 'date',
      time: 'time',
      venue: 'venue',
      ticketUrl: 'url'
    };
    const mockStore = {events: [event1], artistInLocation: [{artist: 'artist'}]};
    const mapped = mapStateToProps(mockStore);

    expect(mapped.events).toEqual(mockStore.events);
    expect(mapped.artistInLocation).toEqual(mockStore.artistInLocation);
  });

});