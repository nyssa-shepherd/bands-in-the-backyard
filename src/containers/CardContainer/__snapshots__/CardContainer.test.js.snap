// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer snapshot test 1`] = `undefined`;
